src/rubicon/objc/api.py

```diff
@@ -91,9 +91,47 @@
 # the Python objects are not destroyed if they are otherwise no Python references left.
 _keep_alive_objects = {}
 
-# Methods that return an object with is implicitly retained by the caller.
-# See https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmRules.html
-_OWNERSHIP_METHOD_PREFIXES = (b"alloc", b"new", b"copy", b"mutableCopy")
+# Methods that return an object which is implicitly retained by the caller.
+# See https://clang.llvm.org/docs/AutomaticReferenceCounting.html#semantics-of-method-families.
+_RETURNS_RETAINED_FAMILIES = {"init", "alloc", "new", "copy", "mutableCopy"}
+
+
+def get_method_family(method_name: str) -> str:
+    """Returns the method family from the method name. See
+    https://clang.llvm.org/docs/AutomaticReferenceCounting.html#method-families for
+    documentation on method families and corresponding selector names."""
+    method_name = method_name.lstrip("_").split(":")[0]
+    leading_lowercases = []
+    for c in method_name:
+        if c.isupper():
+            break
```

Comment:
Darn, thanks for catching that! Reading the ARC docs again, it looks like we have no option than to explicitly check for the method name starting with a family string:

> A selector is in a certain selector family if, ignoring any leading underscores, the first component of the selector either consists entirely of the name of the method family or it begins with that name followed by a character other than a lowercase letter

Reply:
> To cover all the special families, there should also be a test that reference counting works correctly on an object returned from alloc which was never initialized.

Technically, Rubicon treats `alloc` no different than `new`, `copy`, etc. It has only special handling for `init`.

But agreed, it is good to test `alloc` in isolation because it is special on the Objective-C side in not returning an initialized object.

Added a test case now.

Reply:
For completeness I think there should also be a functional test of `mutableCopy`.

Reply:
And `new`. Maybe the common code between these tests could be factored out.

Reply:
I guess that would also mean including a test for `copy` when a new object is returned? Sure, I can add all of those. Factoring out common code would indeed be nice, but may be tricky because we are testing garbage collection and deallocation.

Something to look at tomorrow.

Reply:
Done.